File: backend/app/schemas/resource.py
from uuid import UUID

from pydantic import BaseModel


class VideoResourceResponse(BaseModel):
    id: UUID
    source: str
    search_query: str
    video_id: str
    title: str
    channel_name: str
    thumbnail_url: str | None
    view_count: int | None
    like_count: int | None
    duration: str | None
    relevance_score: float | None
    url: str

    model_config = {"from_attributes": True}

File: frontend/src/types/resource.ts
export interface VideoResource {
  id: string
  source: string
  search_query: string
  video_id: string
  title: string
  channel_name: string
  thumbnail_url: string | null
  view_count: number | null
  like_count: number | null
  duration: string | null
  relevance_score: number | null
  url: string
}
